lib/proc.c

```diff
@@ -342,6 +342,10 @@ static int procusage_cb(pid_t pid __attribute__((unused)),
     /* we only count logged in sessions */
     if (!userid) return 0;
 
+    /* only check for logins to the particular protocol */
+    if (limitsp->servicename && strcmp(servicename, limitsp->servicename))
```

Comment:
nit: Is `servicename` ever allowed to be NULL?

Reply:
it's not called with it being NULL on this codepath, but you could in theory ask for a total set of ALL the connections that a user has.  We do that in another case: when we're killing all user connections.